### ISSUE TITLE
Define Location of Result Panels

### ISSUE THREAD

**Issue by pmcl77** [cweijan/vscode-mysql#472]
I could not find this setting in the extension settings, I am not sure if it is possible with  vscode settings. The location of new result panels seem a bit arbitrary. Sometimes it opens below the editor (horizontal split) and sometimes it appears to the right side of the editor (vertical split). Especially I noticed that when I have locked panels, it will split vertically while with no locked panels it's mostly a horizontal split.

Could be a good feature to have options where result panels are supposed to pop up:
- below editor
- right of editor
- (above editor)
- (left of editor)

**Comment by cweijan** [comment]
vscode does not provide a relevant API to determine the position of the editor, only the horizontal and vertical switching of the split screen, which is currently switched when the query result is rendered, so what you said is impossible.

**Comment by pmcl77** [comment]
Was it possible then to at least define the preferred split, either horizontal or vertical?

**Comment by cweijan** [comment]
I'm just saying vscode doesn't provide an API.

**Comment by RobertOstermann** [comment]
I agree with @pmcl77 that it should be possible to choose the preferred split of the results view. Before version 4.5.0 the results view was always vertical for me, but now always show horizontally. I have created a PR that would fix this issue and allow the user to set the preferred split, @cweijan let me know what you think.

**Comment by cweijan** [comment]
@RobertOstermann  Thanks, I can add this option.

**Comment by pmcl77** [comment]
@RobertOstermann I actually noticed the following behaviour:

- Initially, after running the first query, the panel shows below (vertical split).
- Without locking the first panel, and running another query, the first panel is replaced, the vertical split stays.
- When locking the first panel, and then running another query, the second panel is added, but the split changes to horizontal.
- Leaving both panels as is, running a third query will replace the second panel (since it was not locked), it stays in horizontal split.
- after either locking the second panel or removing it, and then run another query, the new panel is added and the split changes back to vertical.

In Summary it seems to behave like this:
- If by running a query a new panel is **added** -> the split changes. 
- If a present panel is **replaced** by a new query panel -> the split stays as it was.

**Comment by cweijan** [comment]
@pmcl77  Yes, when a new panel is created it will toggle horizontal/vertical, vscode only provides such an API.

**Comment by pmcl77** [comment]
Is there any way this can be changed like @RobertOstermann suggests? I'm starting to get crazy :D

**Comment by cweijan** [comment]
I have published the settings he provided to the latest version.
![image](https://user-images.githubusercontent.com/27798227/157180488-f3093afb-855b-4c2a-b41d-72edb6e64953.png)

**Comment by pmcl77** [comment]
ok, deactivating toggle helps! You said, there is no way to define where the new window will open? It is now opening in vertical split view (i.e. to the right). I remember in an earlier version,  it actually did not open in split view at all, which is what I personally prefer, because my screen space is limited. Any way to do this other than having to manually unsplit it every time?

**Comment by cweijan** [comment]
Ok, I can add an option in the next version.

**Comment by pmcl77** [comment]
Cool, ideally, the options would be

SQL Result Panel Location
- No Split (new tab in current)
- Split right
- Split down
- (Split left)
- (Split up)

I think like this, everyone should have an option that matches their workflow.

**Comment by cweijan** [comment]
I will add a new option in the case of saving toggleLayoutWhenQuery, as I have said above, vscode does not provide an API for Split down.

**Comment by cweijan** [comment]
I have released version 4.7.7 which adds splitEditorWhenQuery option, you can avoid split view by turning it off.
![image](https://user-images.githubusercontent.com/27798227/157417175-395ecff1-31b8-48ba-8b56-cf5892fc0208.png)

**Comment by pmcl77** [comment]
Perfect!
